sycl/doc/design/NonReusableDeviceCode.md

```diff
@@ -0,0 +1,45 @@
+# Non-Reusable Device Code
+
+## Overview
+By default, SYCL allows device code to be reusable, where function calls outside
+of the current translation unit are allowed using the `SYCL_EXTERNAL` attribute.
+To implement this, the compiler linkes all device code together, resolving all
+dependencies during the device linking process. This results in one large LLVM
+module that subsequent steps of device linking use. For more information,
+see [Compiler And Runtime Design](https://github.com/intel/llvm/blob/sycl/sycl/doc/design/CompilerAndRuntimeDesign.md).
+
+The tradeoff with this apporach is that having a single module may increase
+memory usage or run time of the compiler.
+
+In the case where device code does not have any `SYCL_EXTERNAL` functions,
+the `-fno-sycl-rdc` flag may be useful. With this flag specified, the compiler
+will not link all device code into one large module. Instead, it will link
+device code on a per-translation-unit basis,
+which may improve runtime and memory usage of the compiler.
+
+## Device code linking with source files
+Let's take an example where the device compiler has two input files:
+a.cpp and b.cpp, both containing device code. In this case,
```

Comment:
I'm confused about the "no-rdc" mode now.  When in this mode, does `sycl-post-link` run during the compilation step are during the link step?

If it is during the compilation step, that should be called out in this document, because that is different from the way it work in regular (rdc) mode.

It it is during the link step, I'm confused by this section because it says the input files are "a.cpp" and "b.cpp" (i.e. source files).  Should these be "a.o" and "b.o" (i.e. object files) instead?

Reply:
During link, the same as today

Sure, let me reword it, thanks